textcounter/src/main/java/com/github/premnirmal/textcounter/CounterView.java

```diff
@@ -0,0 +1,339 @@
+     *  @param context - context for CounterView constructor
+     *  @param attrs - attributes
+     *  @param defStyleAttr - defStyle attribute
+     * @param defStyleRes - defStyleRes attribute
+     */
+    public CounterView(Context context, AttrSet attrs, String defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr);
+        mHandler = new EventHandler(EventRunner.getMainEventRunner());
+        init(context, attrs, defStyleAttr, defStyleRes);
+        setBindStateChangedListener(this);
+    }
+
+    protected void init(Context context, AttrSet attrs, String defStyleAttr, int defStyleRes) {
+        if (attrs == null) {
+            this.prefix = "";
+            this.suffix = "";
+            this.text = "";
+            this.timeInterval = DEFAULT_INTERVAL;
+            this.increment = DEFAULT_INCREMENT;
+            this.startValue = 0f;
+            this.endValue = 0f;
+            this.autoStart = false;
+            this.autoFormat = true;
+            this.counterType = CounterType.NUMBER;
+            return;
+        }
+        try {
+            final CharSequence prefix1 = AttrUtils.getStringFromAttr(attrs, "prefix");
+            if (prefix1 != null) {
+                this.prefix = prefix1.toString();
+            } else {
+                this.prefix = "";
+            }
+            final CharSequence suffix1 = AttrUtils.getStringFromAttr(attrs, "suffix");
+            if (suffix1 != null) {
+                this.suffix = suffix1.toString();
+            } else {
+                this.suffix = "";
+            }
+            final CharSequence text1 = AttrUtils.getCharSequenceFromAttr(attrs, "ohos_text");
+            if (text1 != null) {
+                this.text = text1.toString();
+            } else {
+                this.text = "";
+            }
+            this.timeInterval = (long) AttrUtils.getFloatFromAttr(attrs, "timeInterval", DEFAULT_INTERVAL);
+            this.increment = AttrUtils.getFloatFromAttr(attrs, "incrementValue", DEFAULT_INCREMENT);
+            this.startValue = AttrUtils.getFloatFromAttr(attrs, "startValue", 0f);
+            this.endValue = AttrUtils.getFloatFromAttr(attrs, "endValue", 0f);
+            this.autoStart = AttrUtils.getBooleanFromAttr(attrs, "autoStart", true);
+            this.autoFormat = AttrUtils.getBooleanFromAttr(attrs, "formatText", true);
+            final int type = getHandletype(AttrUtils.getStringFromAttr(attrs, "type", ""));
+            switch (type) {
+                case 0:
+                    counterType = CounterType.NUMBER;
+                    break;
+                case 1:
+                    counterType = CounterType.DECIMAL;
+                    break;
+                case 2:
+                    counterType = CounterType.BOTH;
+                    break;
+                default:
+                    HiLog.debug(HI_LOG_LABEL, "default switch case in init");
+            }
+        } finally {
+            //finally block
```

Comment:
Can remove this finalyy block, as there is no implementation present in the block

Reply:
Done